File: reference-code/src/main/java/createcommands/arguments/ListedArguments.java
package createcommands.arguments;

import dev.jorel.commandapi.CommandAPICommand;
import dev.jorel.commandapi.arguments.GreedyStringArgument;
import dev.jorel.commandapi.arguments.IntegerArgument;
import dev.jorel.commandapi.arguments.PlayerArgument;
import org.bukkit.entity.Player;

class ListedArguments {
    {
        // #region listedArgumentsExample
        new CommandAPICommand("mycommand")
            .withArguments(new PlayerArgument("player"))
            .withArguments(new IntegerArgument("value").setListed(false))
            .withArguments(new GreedyStringArgument("message"))
            .executes((sender, args) -> {
                // args == [player, message]
                Player player = (Player) args.get("player");
                String message = (String) args.get("message"); // Note that the IntegerArgument is not available in the CommandArguments
                player.sendMessage(message);
            })
            .register();
        // #endregion listedArgumentsExample
    }
}

File: reference-code/src/main/kotlin/createcommands/arguments/ListedArguments.kt
package createcommands.arguments

import dev.jorel.commandapi.CommandAPICommand
import dev.jorel.commandapi.arguments.GreedyStringArgument
import dev.jorel.commandapi.arguments.IntegerArgument
import dev.jorel.commandapi.arguments.PlayerArgument
import dev.jorel.commandapi.executors.CommandExecutor
import dev.jorel.commandapi.kotlindsl.anyExecutor
import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.greedyStringArgument
import dev.jorel.commandapi.kotlindsl.integerArgument
import dev.jorel.commandapi.kotlindsl.playerArgument
import org.bukkit.entity.Player

fun listedArguments() {
    // #region listedArgumentsExample
    CommandAPICommand("mycommand")
        .withArguments(PlayerArgument("player"))
        .withArguments(IntegerArgument("value").setListed(false))
        .withArguments(GreedyStringArgument("message"))
        .executes(CommandExecutor { _, args ->
            // args == [player, message]
            val player = args["player"] as Player
            val message = args["message"] as String // Note that the IntegerArgument is not available in the CommandArguments
            player.sendMessage(message)
        })
        .register()
    // #endregion listedArgumentsExample
}

fun listedArgumentsDSL() {
    // #region listedArgumentsExampleDSL
    commandAPICommand("mycommand") {
        playerArgument("player")
        integerArgument("value") { isListed = false }
        greedyStringArgument("message")
        anyExecutor { _, args ->
            // args == [player, message]
            val player = args["player"] as Player
            val message = args["message"] as String // Note that the IntegerArgument is not available in the CommandArguments
            player.sendMessage(message)
        }
    }
    // #endregion listedArgumentsExampleDSL
}